rholang/src/main/scala/coop/rchain/rholang/interpreter/Interpreter.scala

```diff
@@ -160,3 +159,22 @@ object Interpreter {
         .mkString("Errors received during evaluation:\n", "\n", "\n")
   }
 }
+
+class ErrorHandlingParser(s: Yylex, sf: java_cup.runtime.SymbolFactory) extends parser(s, sf) {
+  import java_cup.runtime.ComplexSymbolFactory.ComplexSymbol
+  import java_cup.runtime.Symbol
+
+  override def unrecovered_syntax_error(cur_token: Symbol): Unit =
+    throw new SyntaxError(
+      cur_token match {
+        case cs: ComplexSymbol =>
+          s"syntax error(${cs.getName}): ${s
+            .yytext()} at ${cs.getLeft.getLine}:${cs.getLeft.getColumn}-${cs.getRight.getLine}:${cs.getRight.getColumn}"
+        case _ => cur_token.toString()
+      }
+    )
+  override def syntax_error(cur_token: Symbol): Unit             = ()
+  override def report_error(message: String, info: Object): Unit = ()
```

Comment:
those "= ()` overrides are turning previous behavior into noops. Most likely this is wrong. If not, it requires a convincing explanation in a comment on the overrides.

Reply:
Yeah; I carefully reviewed the CUP docs and did all that reasoning in my head about why it should be this way and then didn't write it down. :-/